lms/djangoapps/course_blocks/api.py

```diff
@@ -38,6 +38,15 @@ def get_course_block_access_transformers(user):
             which the block structure is to be transformed.
 
     """
+    if grading:
+        return [
+            library_content.ContentLibraryTransformer(),
+            ContentTypeGateTransformer(),
+            user_partitions.UserPartitionTransformer(),
+            visibility.VisibilityTransformer(),
+            field_data.DateOverrideTransformer(user),
+        ]
+
```

Comment:
Is it possible to keep the list the same in both cases, but pass the `grading` parameter into the `StartDateTransformer` so that it can change its behavior in that case? Or does that break the caching somehow?

Reply:
I'll have to investigate this.